### PR TITLE
'first_tasks'

### PR DESCRIPTION
**Pull request by Drop777**

### CODE REVIEW

File: src/server.js [hunk 1]
```diff
@@ -1,6 +1,11 @@
 const express = require('express');
 const path = require('path');
 const fs = require('fs');
+const bodyParser = require('body-parser');
+const uuid = require('uuid/v4');
+
+const jsonParser = bodyParser.json();
```
Comment (YuriiHoliuk) [inline src/server.js]: there are `express.json()` middleware, no need to install separate library

File: src/server.js [hunk 2]
```diff
@@ -64,4 +69,86 @@ app.get('/', (req, res) => {
...
+app.post('/movies', jsonParser, (req, res) => {
+  const { title, imdbRating, year } = req.body;
+  const newMovie = {
+    imdbRating,
+    title,
+    year,
+    id: id,
+  };
+  const movies = JSON.parse(fs.readFileSync(JSON_PATH));
+  dataWithNewFilm = [...movies, newMovie];
```
Comment (YuriiHoliuk) [inline src/server.js]: will be great to add some empty lines

File: src/server.js [hunk 3]
```diff
@@ -64,4 +69,86 @@ app.get('/', (req, res) => {
...
 
+app.post('/movies', jsonParser, (req, res) => {
+  const { title, imdbRating, year } = req.body;
+  const newMovie = {
+    imdbRating,
+    title,
+    year,
+    id: id,
+  };
+  const movies = JSON.parse(fs.readFileSync(JSON_PATH));
```
Comment (YuriiHoliuk) [inline src/server.js]: Do not use Sync methods, your server will be slow

File: src/server.js (hunk 2 above)
Comment (YuriiHoliuk) [inline src/server.js]: Where is variable declaration?
`movie` -> `film` -> `data`??🤔

File: src/server.js [hunk 4]
```diff
@@ -64,4 +69,86 @@ app.get('/', (req, res) => {
...
+  const { title, imdbRating, year } = req.body;
+  const newDataForMovie = {
+    title,
+    imdbRating,
+    year
+  };
+  const movieWithChange = JSON.parse(fs.readFileSync(JSON_PATH))
+    .map(movie => {
+      if (movie.id === req.params.id) {
+        return ({
```
Comment (YuriiHoliuk) [inline src/server.js]: unneeded parentethes

File: src/server.js [hunk 5]
```diff
@@ -64,4 +69,86 @@ app.get('/', (req, res) => {
...
+})
+
+app.put('/movies/:id', jsonParser, (req, res) => {
+  const { title, imdbRating, year } = req.body;
+  const newDataForMovie = {
+    title,
+    imdbRating,
+    year
+  };
+  const movieWithChange = JSON.parse(fs.readFileSync(JSON_PATH))
```
Comment (YuriiHoliuk) [inline src/server.js]: no Sync

File: src/server.js [hunk 6]
```diff
@@ -64,4 +69,86 @@ app.get('/', (req, res) => {
...
+          ...req.body,
+        })
+      }
+      return movie;
+    });
+  fs.writeFile(JSON_PATH, JSON.stringify(movieWithChange), (err) => {
+    if (err) {
+      res.sendStatus(500);
+    } else {
+      res.json(newDataForMovie);
```
Comment (YuriiHoliuk) [inline src/server.js]: Here, you should send object with id

File: src/server.js [hunk 7]
```diff
@@ -64,4 +69,86 @@ app.get('/', (req, res) => {
...
+          year,
+        })
+      }
+      return movie;
+    });
+  fs.writeFile(JSON_PATH, JSON.stringify(movieWithChange), (err) => {
+    if (err) {
+      res.sendStatus(500);
+    } else {
+      res.json(newDataForMovie);
```
Comment (YuriiHoliuk) [inline src/server.js]: Here, you should send object with id

File: src/server.js (hunk 5 above)
Comment (YuriiHoliuk) [reply src/server.js]: movieWithChange - not the best name

File: src/server.js [hunk 8]
```diff
@@ -64,4 +69,86 @@ app.get('/', (req, res) => {
...
+});
+
+app.patch('/movies/:id', jsonParser, (req, res) => {
+  const newDataForMovie = {
+    ...req.body
+  };
+  const movieWithChange = JSON.parse(fs.readFileSync(JSON_PATH))
+    .map(movie => {
+      if (movie.id === req.params.id) {
+        return ({
```
Comment (YuriiHoliuk) [inline src/server.js]: will be better to use ternary here

**Review by YuriiHoliuk — CHANGES_REQUESTED** [review]
There are no first 4 tasks in PR changes

File: src/server.js [hunk 9]
```diff
@@ -1 +1,154 @@
...
+  fs.writeFile(JSON_PATH, JSON.stringify(dataWithNewFilm), (err) => {
+    if (err) {
+      res.sendStatus(500);
+    } else {
+      res.json(newMovie);
+    }
+  });
+})
+
+app.put('/movies/:id', jsonParser, (req, res) => {
```
Comment (YuriiHoliuk) [inline src/server.js]:
```suggestion
app.put('/movies/:id', express.json(), (req, res) => {
```

**Comment by danheim** [comment]
Зачем парсить каждый метод отдельно, если можно `app.use(express.json())`?

**Review by YuriiHoliuk — APPROVED** [review]
Супер, круто зробив!

**Pull request merged**